week2/index.js

```diff
@@ -0,0 +1,157 @@
+        .then(() => askingLoop());
+}
+
+let updateTodoItem = (filename, todoList, taskNumber, itemText) => {
+    todoList[taskNumber - 1] = {
+        TASK_NUMBER: taskNumber,
+        TODO_TASK: itemText
+    };
+    writeFiles(filename, JSON.stringify(myTodoList))
+        .then(() => console.log("\nDone Updating the task on your TODO list.\n"))
+        .then(() => askingLoop());
+}
+
+let resetTodoItem = (filename, todoList) => {
+    todoList = [];
+}
+
+const readline = require('readline');
+const rl = readline.createInterface(process.stdin, process.stdout);
+
+function askingLoop(data) {
+
+    rl.question('Write your commend > ', function (data) {
+
+        switch (data.toString().trim()) {
+            case 'help':
+                readFiles(FILENAMES.help).then((data) => {
+                    console.log(data);
+                    console.log('\nWrite your commend > ')
+                });
+
+                break;
+            case 'list':
+                console.log(`\n----------------------------------------------------------------------------------\n`);
+                myTodoList.map(task => { console.log(`    ${task.TODO_NUMBER}:${task.TODO_TASK}`) });
+                console.log(`\n----------------------------------------------------------------------------------\n`);
+                break;
+            case 'add':
+
+                rl.question('\nWrite a new task or "Ctrl+C" to exit> ', function (task) {
+                    addTodoItem(FILENAMES.todo, myTodoList, task.toString().trim())
+                    askingLoop();
+                })
+                break;
+            case 'remove':
+                rl.question('\nWrite the task number or "Ctrl+C" to exit> ', function (task) {
+                    removeTodoItem(FILENAMES.todo, myTodoList, task);
+                    askingLoop();
+                })
+                break;
+            case 'update':
+                rl.question('\nWrite the task number or "Ctrl+C" to exit> ', function (task) {
+                    let number = task;
+                    rl.question('\nWrite the new task> ', function (data) {
+                        updateTodoItem(FILENAMES.todo, myTodoList, number, data);
+                    })
+
+                })
+
+                break;
+            case 'reset':
+                rl.question('\n are you sure you wanna reset your list (Y/N)> ', function (task) {
+                    if (task.toString().trim() == "Y") {
+                        resetTodoItem(FILENAMES.todo, myTodoList);
+                    }
+                })
+
+                break;
+            case 'exit':
+                console.log('Bye !!.');
+                console.log.exit()
+                break;
+            default:
+                console.log('Sorry, Wrong input kindly use "help" command for more information');
+
+        }
+        askingLoop(data)
+    });
+
+}
+
+readFiles(FILENAMES.todo).then((data) => {

```

Comment:
You have no `catch()` for the promise rejection here, which is why it's throwing the error I posted in a comment.